src/Cli/Microsoft.TemplateEngine.Cli/Commands/InteractiveMode/InteractiveMode.cs

```diff
@@ -0,0 +1,253 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Edge;
+using Microsoft.TemplateEngine.Edge.Settings;
+
+namespace Microsoft.TemplateEngine.Cli.Commands.InteractiveMode
+{
+    public static class InteractiveMode
+    {
+        public static async Task EnterInteractiveMode(InvocationContext context, Func<InvocationContext, Task> next)
+        {
+            // TODO: make sure we have access to Instantiate command
+            if (context.ParseResult.CommandResult.Command.Name == "new" || context.ParseResult.CommandResult.Command.Name == "create")
```

Comment:
`is NewCommand` and `is InstantiateCommand` is more reliable way as in parsing tree might be other commands named `create`.